front/pages/w/[wId]/members/index.tsx

```diff
@@ -569,17 +561,23 @@ function ChangeMemberModal({
   owner: WorkspaceType;
 }) {
   const { mutate } = useSWRConfig();
+  const [currentRole, setCurrentRole] = useState<RoleType | null>(null);
+  const [revokeMemberModalOpen, setRevokeMemberModalOpen] = useState(false);
   if (!member) return null; // Unreachable
   const roleTexts: { [k: string]: string } = {
     admin: "Admins can manage members, in addition to builders' rights.",
     builder:
       "Builders can create custom assistants and use advanced dev tools.",
     user: "Users can use assistants provided by Dust as well as custom assistants created by their company.",
   };
+  const onCloseMutate = async () => {
```

Comment:
Why do we mutate on close? What if we didn't change anything in the modal?
It seems more logic before when we were mutating on `handleMemberRoleChange`.

Reply:
* If we didn't change anything the mutate has no effect and is not perceptible by the user (tested)
* Mutating in handlemberrolechange closes the modal, I didn't find a way to avoid this-
(note: keeping the modal open was one of the goal of this PR)
but if there's a simple way I missed I'm glad to change since I also would have preferred it this way

Reply:
Why is the modal geting closed?

We probably want to fix the root cause here.

Reply:
What is the final strategy here? And what was the root cause ooc?